tests/apps/test_graph_similarity.py

```diff
@@ -89,3 +92,113 @@ def test_sample_to_event(dim, max_count_per_mode):
+        samp = similarity.orbit_to_sample(all_orbits[orb_dim][0], modes_dim)
+        assert len(samp) == modes_dim
+
+    def test_sample_composition(self):
+        """Test if function returns a sample that corresponds to the input orbit. Input orbits
+        are orbits from ``all_orbits_cumulative``, i.e., all orbits from 3-5 photons. This test
+        checks if a sample corresponds to an orbit by counting the occurrence of elements in the
+        sample and comparing to a count of elements in the orbit."""
+        modes = 5
+
+        all_orbits_zeros = [
+            [1, 1, 1, 0, 0],
+            [2, 1, 0, 0, 0],
+            [3, 0, 0, 0, 0],
+            [1, 1, 1, 1, 0],
+            [2, 1, 1, 0, 0],
+            [3, 1, 0, 0, 0],
+            [2, 2, 0, 0, 0],
+            [4, 0, 0, 0, 0],
+            [1, 1, 1, 1, 1],
+            [2, 1, 1, 1, 0],
+            [3, 1, 1, 0, 0],
+            [2, 2, 1, 0, 0],
+            [4, 1, 0, 0, 0],
+            [3, 2, 0, 0, 0],
+            [5, 0, 0, 0, 0],
+        ]  # padding orbits with zeros at the end for comparison to samples
+
+        counts = [Counter(similarity.orbit_to_sample(o, modes)) for o in all_orbits_cumulative]
+        ideal_counts = [Counter(o) for o in all_orbits_zeros]
+
+        assert counts == ideal_counts
+
+
+class TestEventToSample:
+    """Tests for the function ``strawberryfields.apps.graph.similarity.event_to_sample``"""
+
+    def test_low_count(self):
+        """Test if function raises a ``ValueError`` if ``max_count_per_mode`` is not positive."""
+        with pytest.raises(ValueError, match="Maximum number of photons"):
+            similarity.event_to_sample(2, 0, 5)
+
+    def test_high_photon(self):
+        """Test if function raises a ``ValueError`` if ``photon_number`` is so high that it
+        cannot correspond to a sample given the constraints of ``max_count_per_mode`` and
+        ``modes``"""
+        with pytest.raises(ValueError, match="No valid samples can be generated."):
+            similarity.event_to_sample(5, 1, 4)
+
+    @pytest.mark.parametrize("photon_num", [3, 4, 5, 6])
+    @pytest.mark.parametrize("modes_dim", [10, 11])
+    @pytest.mark.parametrize("count", [3, 4, 5, 6])
+    def test_sample_length(self, photon_num, modes_dim, count):
+        """Test if function returns a sample that is of correct length ``modes_dim``."""
+        samp = similarity.event_to_sample(
+            photon_number=photon_num, max_count_per_mode=count, modes=modes_dim
+        )
+        assert len(samp) == modes_dim
+
+    @pytest.mark.parametrize("photon_num", [3, 4, 5, 6])
+    @pytest.mark.parametrize("modes_dim", [10, 11])
+    @pytest.mark.parametrize("count", [3, 4, 5, 6])
+    def test_sample_sum(self, photon_num, modes_dim, count):
+        """Test if function returns a sample that has the correct number of photons."""
+        samp = similarity.event_to_sample(
+            photon_number=photon_num, max_count_per_mode=count, modes=modes_dim
```

Comment:
Not a big deal but, do we have to use `photon_number=photon_num, max_count_per_mode=count, modes=modes_dim` instead of just `photon_num, count, modes_dim` ?

Reply:
I'm in favour of less verbose argument names, even if it means users will have to double check the docstring for queries.

Reply:
Have made shorter 😁